### PR TITLE
feat (api): crud project key

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: engine/api/keys/keys.go [hunk 1]
```diff
@@ -35,3 +46,53 @@ func Generatekeypair(keyname string) (string, string, error) {
 
 	return pub, priv, err
 }
+
+// GenerateGPGKeyPair generates a private / public PGP key
```
Comment (yesnault) [inline engine/api/keys/keys.go]: doc GeneratePGPKeyPair

File: cli/cdsctl/project_key.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/spf13/cobra"
+
+	"github.com/ovh/cds/cli"
+	"github.com/ovh/cds/sdk"
+)
+
+var (
+	projectKeyCmd = cli.Command{
+		Name:  "key",
+		Short: "Manage CDS project key",
```
Comment (yesnault) [inline cli/cdsctl/project_key.go]: keys

File: cli/cdsctl/project_key.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/ovh/cds/cli"
+	"github.com/ovh/cds/sdk"
+)
+
+var (
+	projectKeyCmd = cli.Command{
+		Name:  "key",
```
Comment (yesnault) [inline cli/cdsctl/project_key.go]: keys

File: sdk/cdsclient/client_project_key.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+import (
+	"fmt"
+
+	"github.com/ovh/cds/sdk"
+)
+
+func (c *client) ProjectKeysList(key string) ([]sdk.ProjectKey, error) {
+	k := []sdk.ProjectKey{}
+	code, err := c.GetJSON("/project/"+key+"/keys", &k)
```
Comment (yesnault) [inline sdk/cdsclient/client_project_key.go]: escape key ?

File: engine/api/main_routes.go [hunk 5]
```diff
@@ -72,6 +72,8 @@ func (router *Router) init() {
 	router.Handle("/project/{permProjectKey}/variable/{name}/audit", GET(getVariableAuditInProjectHandler))
 	router.Handle("/project/{permProjectKey}/applications", GET(getApplicationsHandler), POST(addApplicationHandler))
 	router.Handle("/project/{permProjectKey}/notifications", GET(getProjectNotificationsHandler))
+	router.Handle("/project/{permProjectKey}/keys", GET(getKeysInProjectHandler), POST(addKeyInProjectHandler))
```
Comment (yesnault) [inline engine/api/main_routes.go]: no update - no PUT?

File: engine/api/project_key.go [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+	p, errP := project.Load(db, key, c.User)
+	if errP != nil {
+		return sdk.WrapError(errP, "addKeyInProjectHandler> Cannot load project")
+	}
+	newKey.ProjectID = p.ID
+
+	switch newKey.Type {
+	case sdk.KeyTypeSsh:
+		pub, priv, errGenerate := keys.Generatekeypair(newKey.Name)
```
Comment (yesnault) [inline engine/api/project_key.go]: no check of newKey.Name ? space allowed?

File: engine/api/project_key.go [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+	vars := mux.Vars(r)
+	key := vars["permProjectKey"]
+
+	p, errP := project.Load(db, key, c.User)
+	if errP != nil {
+		return sdk.WrapError(errP, "getKeysInProjectHandler> Cannot load project")
+	}
+
+	if errK := project.LoadAllKeys(db, p); errK != nil {
+		return sdk.WrapError(errP, "getKeysInProjectHandler> Cannot load project keys")
```
Comment (yesnault) [inline engine/api/project_key.go]: errK

File: engine/api/main_routes.go (hunk 5 above)
Comment (sguiheux) [reply engine/api/main_routes.go]: no

File: sdk/error.go [hunk 8]
```diff
@@ -217,6 +218,7 @@ var errorsAmericanEnglish = map[int]string{
 	ErrInvalidJobRequirement.ID:                 "Invalid job requirement",
 	ErrNotImplemented.ID:                        "This functionality isn't implemented",
 	ErrParameterNotExists.ID:                    "This parameter doesn't exist",
+	ErrUnknownKeyType.ID:                        "Unkown key type",
```
Comment (bnjjj) [inline sdk/error.go]: unknown

File: engine/api/project_key.go (hunk 6 above)
Comment (sguiheux) [reply engine/api/project_key.go]: yes

File: sdk/cdsclient/client_project_key.go (hunk 4 above)
Comment (sguiheux) [reply sdk/cdsclient/client_project_key.go]: it's the project key so no :)

**Pull request merged**